src/contracts/Chef.sol

```diff
@@ -1,203 +1,45 @@
+abstract contract Chef is Ownable {
+    uint256 private _metricPerBlock;
     uint256 public constant ACC_METRIC_PRECISION = 1e12;
 
     bool private _rewardsActive;
-    AllocationGroup[] private _allocations;
-    uint256 private _totalAllocPoint;
-    uint256 private _lifetimeShareValue = 0;
-    uint256 private _lastRewardBlock;
+    uint256 private _lastRewardBlock; 
 
     MetricToken private _metric;
 
-    constructor(address metricTokenAddress) {
-        _metric = MetricToken(metricTokenAddress);
-        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
-        _grantRole(ALLOCATION_ROLE, msg.sender);
-    }
-
-    //------------------------------------------------------Manage Allocation Groups
-
-    function addAllocationGroup(
-        address newAddress,
-        uint256 newShares,
-        bool newAutoDistribute
-    ) external onlyRole(ALLOCATION_ROLE) nonDuplicated(newAddress) {
-        if (_rewardsActive && _totalAllocPoint > 0) {
-            updateAccumulatedAllocations();
-        }
-        AllocationGroup memory group = AllocationGroup({
-            groupAddress: newAddress,
-            shares: newShares,
-            autodistribute: newAutoDistribute,
-            rewardDebt: newShares.mul(_lifetimeShareValue).div(ACC_METRIC_PRECISION),
-            claimable: 0
-        });
-
-        _allocations.push(group);
-        _totalAllocPoint = _totalAllocPoint.add(group.shares);
-    }
-
-    function updateAllocationGroup(
-        address groupAddress,
-        uint256 agIndex,
-        uint256 shares,
-        bool newAutoDistribute
-    ) public onlyRole(ALLOCATION_ROLE) {
-        if (_rewardsActive && _totalAllocPoint > 0) {
-            updateAccumulatedAllocations();
-        }
-        _totalAllocPoint = _totalAllocPoint.sub(_allocations[agIndex].shares).add(shares);
-        _allocations[agIndex].groupAddress = groupAddress;
-        _allocations[agIndex].shares = shares;
-        _allocations[agIndex].autodistribute = newAutoDistribute;
-    }
-
-    function removeAllocationGroup(uint256 agIndex) external onlyRole(ALLOCATION_ROLE) {
-        require(agIndex < _allocations.length);
-        if (_rewardsActive && _totalAllocPoint > 0) {
-            updateAccumulatedAllocations();
-        }
-        _totalAllocPoint = _totalAllocPoint.sub(_allocations[agIndex].shares);
-
-        _allocations[agIndex] = _allocations[_allocations.length - 1];
-        _allocations.pop();
-    }
+    //------------------------------------------------------Setters
 
-    function toggleRewards(bool isOn) external onlyRole(ALLOCATION_ROLE) {
+    function toggleRewards(bool isOn) public onlyOwner() {
         _rewardsActive = isOn;
         _lastRewardBlock = block.number;
     }
 
-    //------------------------------------------------------Getters
-
-    function getAllocationGroups() public view returns (AllocationGroup[] memory) {
-        return _allocations;
+    function setMetricPerBlock(uint256 metricAmount) public virtual {
```

Comment:
should this have the onlyOwner modifier?

Reply:
I think so, don't want just anyone to be able to call this